tensorflow/tensorflow.bzl

```diff
@@ -910,28 +921,50 @@ def _cuda_copts():
       ]),
   })
 
+def _rocm_copts():
+  """Gets the appropriate set of copts for (maybe) ROCm compilation.
+
+    If we're doing ROCm compilation, returns copts for our particular ROCm
+    compiler.  If we're not doing ROCm compilation, returns an empty list.
+
+    """
+  return rocm_default_copts() + select({
+      "//conditions:default": [],
+      "@local_config_rocm//rocm:using_hipcc": ([
+          "",
+      ]),
+  })
+
 # Build defs for TensorFlow kernels
 
 # When this target is built using --config=cuda, a cc_library is built
 # that passes -DGOOGLE_CUDA=1 and '-x cuda', linking in additional
 # libraries needed by GPU kernels.
+#
+# When this target is built using --config=rocm, a cc_library is built
+# that passes -DTENSORFLOW_USE_ROCM and '-x rocm', linking in additional
+# libraries needed by GPU kernels.
 def tf_gpu_kernel_library(srcs,
                           copts=[],
                           cuda_copts=[],
                           deps=[],
                           hdrs=[],
                           **kwargs):
-  copts = copts + _cuda_copts() + if_cuda(cuda_copts) + tf_copts()
+  copts=copts + tf_copts() + _cuda_copts() + _rocm_copts() + if_cuda_is_configured(cuda_copts) + if_rocm_is_configured(cuda_copts)
```

Comment:
let's hide the additional complexities here by adding an arg to "_cuda_copts" and "_rocm_copts"

So this line becomes:
copts=copts + tf_copts() + _cuda_copts(opts=cuda_copts) + _rocm_copts(opts=cuda_copts)

Reply:
@gunan Thanks. I'll address this.